Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptions.swift

```diff
@@ -4,120 +4,93 @@
 
 import Foundation
 
-/// Describes the layout of base message content view.
-public struct ChatMessageLayoutOptions: OptionSet, Hashable {
-    public let rawValue: Int
+/// A typealias of `Set<ChatMessageLayoutOption>` to make the API similar of an `OptionSet`.
+public typealias ChatMessageLayoutOptions = Set<ChatMessageLayoutOption>
 
-    public init(rawValue: Int) {
+public extension ChatMessageLayoutOptions {
+    /// The raw value of all the options that is used to create identify a collection of options.
+    /// It is essentially to make the API backwards-compatible with `OptionSet`
+    /// and used to create the reuse identifier of the message cell.
+    var rawValue: String {
+        // Since it is a Set, we need to sort it to make sure the identifier doesn't change.
+        map(\.rawValue).sorted().joined(separator: "-")
+    }
+
+    mutating func remove(_ options: ChatMessageLayoutOptions) {
+        self = subtracting(options)
+    }
+    
+    mutating func insert(_ options: ChatMessageLayoutOptions) {
+        options.forEach { self.insert($0) }
+    }
+}
+
+/// Each message layout option is used to define which views will be part of the message cell.
+/// A different combination of layout options will produce a different cell reuse identifier.
+public struct ChatMessageLayoutOption: RawRepresentable, Hashable, ExpressibleByStringLiteral {
+    public let rawValue: String
+
+    public init(rawValue: String) {
         self.rawValue = rawValue
     }
+
+    public init(stringLiteral value: String) {
+        self.init(rawValue: value)
+    }
+
+    // Probably this is not needed. It will be a breaking change anyway to the current customer
+    // since the customer was doing initialising from `ChatMessageLayoutOptions`.
+    //
+//    @available(*, deprecated, message: "Use the string raw value initialiser.")
+//    public init(rawValue: Int) {
+//        self.rawValue = "\(rawValue)"
+//    }
```

Comment:
I'm planning on removing this. Because this will be a breaking API anyways :/ Since customers were initialising on `ChatMessageLayoutOptions` not in `ChatMessageLayoutOption`